src/core/thread/mle_router.cpp

```diff
@@ -935,24 +936,28 @@ Error MleRouter::HandleLinkAccept(RxInfo &aRxInfo, bool aRequest)
         }
 
         // Route (optional)
-        switch (error = ProcessRouteTlv(aRxInfo, routeTlv))
+        switch (aRxInfo.mMessage.ReadRouteTlv(routeTlv))
         {
         case kErrorNone:
+            VerifyOrExit(routeTlv.IsRouterIdSet(routerId), error = kErrorParse);
+
+            if (mRouterTable.IsRouteTlvIdSequenceMoreRecent(routeTlv))
```

Comment:
I was initially considering doing the same but decided to keep up separate, since we can call `ProcessRouteTlv()`  from different places (and may add new places to call it from in future)
- e.g., from messages received on FED or when we attach first time
-  I was not sure adding this check would be correct in all cases (it may be ok/ignored in most cases but I felt it would be risky to include it in `ProcessRouteTlv()`).